packages/esm-ohri-pmtct-app/src/post-submission-actions/art-linkage-action.ts

```diff
@@ -7,24 +7,23 @@ const ArtSubmissionAction: PostSubmissionAction = {
   applyAction: async function ({ patient, encounters, sessionMode }) {
     const encounter = encounters[0];
     const encounterLocation = encounter.location['uuid'];
-    // Only allow this action in enter mode
+
     if (sessionMode !== 'enter') {
       return;
     }
-    // save Art Number to patient identifiers
+
     let artNumber = encounter.obs.find((observation) => observation.concept.uuid === artNoConcept)?.value;
     if (!artNumber) {
       return;
     }
-
-    // convert art number to string
     if (typeof artNumber !== 'string') {
       artNumber = artNumber.toString();
     }
 
+    //Patient can only have one ART No.
     const patientIdentifiers = await fetchPatientIdentifiers(patient.id);
-    const existingArtNumbers = patientIdentifiers.filter((id) => id.identifierType.uuid === artNoConcept);
-    if (existingArtNumbers.some((artNumber) => artNumber.identifier === artNumber)) {
+    const existingArtNumbers = patientIdentifiers.filter((id) => id.identifierType.uuid === artUniqueNumberType);
```

Comment:
Is this another concept different from the one previously specified or was this just a rename?

Reply:
This code was actually wrong but I guess it was never validated. Each identifier type has a unique uuid which has nothing to do with the concept. So at the point of saving we pass the `artUniqueNumberType ` but the point of retrieving we were checking against the `artNoConcept ` which was wrong